### ISSUE TITLE
Cannot deploy custom role referring to custom privilege

### ISSUE THREAD

**Issue by dmcassel** [marklogic-community/ml-gradle#365]
I'm trying to grant a custom URI privilege to a custom role. If I configure them both with the role referring to the privilege and `gradle mldeploysecurity`, I get an error because [roles get created first](https://github.com/marklogic-community/ml-app-deployer/blob/62e165c26f3b75e77cb9d4522c26e0f4af9ec2c9/src/main/java/com/marklogic/appdeployer/command/CommandMapBuilder.java#L68-L75) and the privilege doesn't yet exist. 

> Task :mlDeploySecurity FAILED
Logging HTTP response body to assist with debugging: {"errorResponse":{"statusCode":"500", "status":"Internal Server Error", "messageCode":"SEC-PRIVDNE", "message":"SEC-PRIVDNE: (err:FOER0000) Privilege does not exist: action(/com.marklogic.smart-mastering/options/algorithms), kind(uri)"}}

If I leave the privilege out of the role and include the role in privilege, I get this error: 

> Task :mlDeploySecurity FAILED
Logging HTTP response body to assist with debugging: {"errorResponse":{"statusCode":"400", "status":"Bad Request", "messageCode":"MANAGE-INVALIDCONFIG", "message":"MANAGE-INVALIDCONFIG: (err:FOER0000) Invalid server configuration:  Cannot create privilege in database"}}

The privilege config looks like this: 

    {
      "privilege-name": "algorithms",
      "action": "/com.marklogic.smart-mastering/options/algorithms/",
      "kind": "uri",
      "role": ["mdm-user"]
    }

That config looks right to be from the documentation, but maybe I'm missing something. The only message in the error log is:

> 2018-07-16 18:34:56.022 Info: MANAGE-INVALIDCONFIG: (err:FOER0000) Invalid server configuration:  Cannot create privilege in database
2018-07-16 18:34:56.022 Info: 388:10:/MarkLogic/manage/endpoints/security/privileges.xqy

**Comment by rjrudin** [comment]
The order in which the commands are added to the list doesn't matter - the order is controlled by the "sortOrder" field on each command. Those values are also defined at https://github.com/marklogic-community/ml-app-deployer/blob/master/src/main/java/com/marklogic/appdeployer/command/SortOrderConstants.java 

Privileges are deployed first, and thus can't refer to any roles. But roles can refer to any privileges. 

I just did a quick test locally on a project with custom roles and privileges - I ran mlDeploySecurity, and the privileges were deployed, followed by the roles.
